src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -27,37 +29,16 @@ internal DateTime StartTimeCore
         {
             get
             {
-                // Get the RUsage data and convert the process start time.
-                // To calculcate the process start time in OSX we call proc_pid_rusage() and use ri_proc_start_abstime.
-                //
-                // ri_proc_start_abstime is absolute time which is the time since some reference point. The reference point
-                // usually is the system boot time but this is not necessary true in all versions of OSX. For example in Sierra 
-                // Mac OS 10.12 the reference point is not really the boot time. To be always accurate in our calculations
-                // we don’t assume the reference point to be the boot time and instead we calculate it by calling GetTimebaseInfo
-                // which is a wrapper for native mach_absolute_time(). That method returns the current time referenced to the needed start point
-                // Then we can subtract the returned time from DateTime.UtcNow and we’ll get the exact reference point.
-                //
-                // The absolute time is measured by the bus cycle of the processor which is measured in nanoseconds multiplied 
-                // by some factor “numer / denom”. In most cases the factor is (1 / 1) but we have to get the factor and use it just 
-                // in case we run on a machine has different factor. To get the factor we call GetTimebaseInfo the wrapper for the 
-                // mach_timebase_info() which give us the factor. Then we multiply the factor by the absolute time and the divide
-                // the result by 10^9 to convert it from nanoseconds to seconds.
-
                 EnsureState(State.HaveNonExitedId);
+                Interop.libproc.proc_taskallinfo? info = Interop.libproc.GetProcessInfoById(Id);
 
-                uint numer, denom;
-                Interop.Sys.GetTimebaseInfo(out numer, out denom);
```

Comment:
Can any of this interop support be removed now, e.g. can the interop for GetTimebaseInfo be removed?

Reply:
A quick GH search says yes. It looks like both GetAbsoluteTime and GetTimebaseInfo are not referenced anywhere other than Process.OSX.

Reply:
It goes all the way to System.Native's pal_time.c/pal_time.h. Are these part of corefx, or will this need a coreclr PR as well to remove?

Reply:
That's all in corefx, e.g. https://github.com/dotnet/corefx/blob/master/src/Native/Unix/System.Native/pal_time.c

Reply:
Ok, resected. It appears with that removed `HAVE_MACH_TIMEBASE_INFO` is no longer used as well, so I'll pull that from the cmake config as well.